samples/multi-thread/wasm-apps/main.c

```diff
@@ -45,6 +48,11 @@ main(int argc, char *argv[])
         goto fail1;
     }
 
+    if (!(sem = sem_open("testsem", 0100 | 02))) {
+        printf("Failed to open sem. %p\n", sem);
+        goto fail1;
```

Comment:
fail1 doesn't destroy the cond, here had better change to `goto fail2`, and in L59, change to `goto fail3`:
```C
fail3:
    sem_close(sem);
    sem_unlink("testsem");
fail2:
    pthread_cond_destroy(&cond);
fail1:
    pthread_mutex_destroy(&mutex);
```